pkgs/development/python-modules/mastodon-archive/default.nix

```diff
@@ -0,0 +1,33 @@
+{ lib
+, fetchFromGitHub
+, buildPythonPackage
+, html2text
+, mastodon-py
+, progress
+}:
+
+buildPythonPackage rec {
+  pname = "mastodon-archive";
+  version = "1.3.1";
+
+  src = fetchFromGitHub {
+    owner = "kensanata";
+    repo = "mastodon-backup";
+    rev = "v${version}";
+    sha256 = "1dlrkygywxwm6xbn0pnfwd3f7641wnvxdyb5qihbsf62w1w08x8r";
+  };
+
+  propagatedBuildInputs = [
+    html2text
+    mastodon-py
+    progress
+  ];
+
+  meta = with lib; {
+    description = "Utility for backing up your Mastodon content";
+    homepage = "https://alexschroeder.ch/software/Mastodon_Archive";
+    license = licenses.gpl3Plus;
+    platforms = platforms.all;
```

Comment:
```suggestion
```
That's automatically set for Python packages.

Reply:
@dotlambda, thanks for the review!
- Regarding tests, what is the point of setting `doCheck = false`? AFAIU it would silently skip any future tests instead of testing or warn us.
- Regarding `pythonPackages` that's what I had initialy done but [I was told](https://github.com/NixOS/nixpkgs/pull/116181#discussion_r650608760) to move it to a `<python-module>` and use `toPythonApplication`. Have I misunderstood something? Should I rollback?
- Regarding `python3Packages.mastodon-py`, I don't want anything in particular, I just rebased my initial work (changing `mastodonpy` to https://github.com/NixOS/nixpkgs/pull/126795 's `mastodon-py`). I'm no Python expert, so if I'm told the current packaging (using `fetchFromGitHub` instead of `fetchPypi` and without some `pytest-*` inputs) is better then I'll drop the commit.

Reply:
> * Have I misunderstood something? Should I rollback?

I think @jonringer mistakenly thought it was in `python-packages.nix`, just roll back.

> * Regarding tests, what is the point of setting `doCheck = false`? AFAIU it would silently skip any future tests instead of testing or warn us.

That is a disadvantage indeed, but not adding it might break automatic updates. Also, it's just a thing that we agreed to for all Python packages and we should be consistent.



> * Regarding `python3Packages.mastodon-py`

Having less `checkInputs` reduces the likelihood of breakage and I don't see a disadvantage of `fetchFromGitHub`. You can however add yourself as a maintainer if you want.

Reply:
> just roll back

Actually, please don't use `python3Packages.callPackage`, but pass `python3` as an argument, and put the expression in `pkgs/tools/backups` since it has no GUI.

Reply:
@dotlambda, thanks again, is it like you wish now?